movingpandas/trajectory_cleaner.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+
+import pandas as pd
+from copy import copy
+
+from .trajectory import Trajectory
+from .trajectory_collection import TrajectoryCollection
+
+
+class TrajectoryCleaner:
+    """
+    Cleaner base class
+    """
+
+    def __init__(self, traj):
+        """
+        Create TrajectoryCleaner
+
+        Parameters
+        ----------
+        traj : Trajectory or TrajectoryCollection
+        """
+        self.traj = traj
+
+    def clean(self, features):
+        """
+        Clean the input Trajectory/TrajectoryCollection.
+
+        Parameters
+        ----------
+        features: dictionary
+            Information regarding the features that will be used to clean the trajectory
+            and the accompanying thresholds. ex. - {'speed': 30, 'heading': 360} etc.
+
+        Returns
+        -------
+        Trajectory/TrajectoryCollection
+            Cleaned Trajectory or TrajectoryCollection
+        """
+        if isinstance(self.traj, Trajectory):
+            return self._clean_traj(self.traj, features)
+        elif isinstance(self.traj, TrajectoryCollection):
+            return self._clean_traj_collection(features)
+        else:
+            raise TypeError
+
+    def _clean_traj_collection(self, features):
+        cleaned = []
+        for traj in self.traj:
+            cleaned.append(self._clean_traj(traj, features))
+        result = copy(self.traj)
+        result.trajectories = cleaned
+        return result
+
+    def _clean_traj(self, traj, features):
+        return traj
+
+
+class OutlierCleaner(TrajectoryCleaner):
+    """
+    Outlier (interquantile range - iqr) based cleaner.
+
+    features : dictionary
+        Key - value pairs of features and alpha (iqr multiplier).
```

Comment:
If I understand this correctly, it operates on existing columns in the dataframe and does not compute any new features that could be useful to identify trajectory outliers. Is that correct?

Reply:
Yes exactly

Reply:
In GIS, the term "feature" usually stands for an object (its geometry and attributes). I'm therefore wondering if we can improve the terminology to be more consistent with what we already have, e.g. in https://movingpandas.readthedocs.io/en/master/trajectory.html and https://movingpandas.readthedocs.io/en/master/trajectorycollection.html. For example, the TrajectoryCollection docs just talk about dataframe columns.:

> get_max(column)
> Return maximum value in the provided DataFrame column over all trajectories

Reply:
OK then, I will swap all the feature terminology :)

Reply:
Done